crates/composer_api/src/lib.rs

```diff
@@ -15,8 +15,16 @@ pub const DEFAULT_SERVER_ADDRESS: &str = "localhost:8888";
 #[derive(Serialize, Deserialize, Debug)]
 pub enum Event {
     TestTick,
-    StdoutWrite { length: usize },
-    StderrWrite { length: usize },
+    TimedTestTick {
+        /// timestamp as Duration since the UNIX_EPOCH
+        ts: Duration,
+    },
+    StdoutWrite {
+        length: usize,
+    },
+    StderrWrite {
+        length: usize,
+    },
     LogStats(LogStats),
 }
```

Comment:
I'm not sure if the timestamp should be encapsulated in (some) Events or of we should send it alongside the events eg. in a wrapping struct.

To me, the later seems nicer but it doesn't fit nicely with sending multiple timestamped events in a single message which we do in the [log prove](https://github.com/tonarino/acoustic_profiler/pull/38/files#diff-dc41a8d6981cf8fc8a37dfe62d23c2d48692271f844aefb407aaa7b9a1fa4486R30-R34).

Reply:
> To me, the later seems nicer

To me too. I even starter to write a comment above saying: At some point (not in this PR) we may want to add timestamp to all events, something like
```rust
pub struct Event {
    timestamp: Duration,
    kind: EventKind,
}

pub enum EventKind {
    TestTick,
    ...
}
```

> But it doesn't fit nicely with sending multiple timestamped events in a single message which we do in the [log probe](https://github.com/tonarino/acoustic_profiler/pull/38/files#diff-dc41a8d6981cf8fc8a37dfe62d23c2d48692271f844aefb407aaa7b9a1fa4486R30-R34).

Hmm, maybe we should have 2 distinct "higher-kind" event types, OneShot and Aggregate/Continuous or something like that. So `struct Event` could again become
```rust
enum Event {
    OneShot { timestmap: ..., kind: OneShotEventKind },
    Aggregate { ... }
}
```

Reply:
I really like how elegantly @skywhale solved this in #41 by *always* sending a Vec of optionally timed Events. `OneShot` is then just a regular message with a single event in it. .)